From 4b1815195dec84140d94cb220c7bf9c7295e702c Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Tue, 19 Aug 2014 15:27:27 -0400
Subject: [tools] perf/record: Refactor feature handling into a separate function

Message-id: <1408462094-14194-279-git-send-email-jolsa@redhat.com>
Patchwork-id: 88250
O-Subject: [PATCH RHEL7.1 BZ1131394 278/325] perf record: Refactor feature handling into a separate function
Bugzilla: 1131394
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1131394
https://bugzilla.redhat.com/show_bug.cgi?id=1131394

upstream
========
commit 57706abc19afc60f0b629af839d2ebee17739f59
Author: David Ahern <dsahern@gmail.com>
Date: Wed Nov 6 11:41:34 2013 -0700

description
===========
Code move only. No logic changes.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/builtin-record.c b/tools/perf/builtin-record.c
index ea4c04f..2932069 100644
--- a/tools/perf/builtin-record.c
+++ b/tools/perf/builtin-record.c
@@ -342,9 +342,28 @@ out:
  return rc;
 }
 
+static void perf_record__init_features(struct perf_record *rec)
+{
+ struct perf_evlist *evsel_list = rec->evlist;
+ struct perf_session *session = rec->session;
+ int feat;
+
+ for (feat = HEADER_FIRST_FEATURE; feat < HEADER_LAST_FEATURE; feat++)
+  perf_header__set_feat(&session->header, feat);
+
+ if (rec->no_buildid)
+  perf_header__clear_feat(&session->header, HEADER_BUILD_ID);
+
+ if (!have_tracepoints(&evsel_list->entries))
+  perf_header__clear_feat(&session->header, HEADER_TRACING_DATA);
+
+ if (!rec->opts.branch_stack)
+  perf_header__clear_feat(&session->header, HEADER_BRANCH_STACK);
+}
+
 static int __cmd_record(struct perf_record *rec, int argc, const char **argv)
 {
- int err, feat;
+ int err;
  unsigned long waking = 0;
  const bool forks = argc > 0;
  struct machine *machine;
@@ -371,17 +390,7 @@ static int __cmd_record(struct perf_record *rec, int argc, const char **argv)
 
  rec->session = session;
 
- for (feat = HEADER_FIRST_FEATURE; feat < HEADER_LAST_FEATURE; feat++)
-  perf_header__set_feat(&session->header, feat);
-
- if (rec->no_buildid)
-  perf_header__clear_feat(&session->header, HEADER_BUILD_ID);
-
- if (!have_tracepoints(&evsel_list->entries))
-  perf_header__clear_feat(&session->header, HEADER_TRACING_DATA);
-
- if (!rec->opts.branch_stack)
-  perf_header__clear_feat(&session->header, HEADER_BRANCH_STACK);
+ perf_record__init_features(rec);
 
  if (forks) {
   err = perf_evlist__prepare_workload(evsel_list, &opts->target,
-- 
1.7.1